R/deployApp.R

```diff
@@ -360,7 +360,7 @@ deployApp <- function(appDir = getwd(),
     stop("Posit Connect does not support deploying without uploading. ",
          "Specify upload=TRUE to upload and re-deploy your application.")
   }
-  if (!isConnectServer(target$server) && length(envVars) > 1) {
+  if (!isConnectServer(target$server) && length(envVars) > 0) {
     cli::cli_abort("{.arg envVars} only supported for Posit Connect servers")
   }
```

Comment:
Rather than checking here by enforcing that `target$server` points to a Connect server, you could instead check just a little bit further down after creating `client`. In that case, you could use a block like the following, and check that `client` implements the `setEnvVars()` psuedo-method.

```r
  client <- clientForAccount(accountDetails)
  if (length(envVars) > 0 && !"setEnvVars" %in% names(client)) {
    cli::cli_abort("{target$server} does not support setting {.arg envVars}")
  }
```

Reply:
I like that idea. Want to do it?

Reply:
No problem, done in f3dac5b38fdf05defe8b65bfd2c2d58eaa488f9b